guides/keras_core/migration_guide_tf_keras_3_0.py

```diff
@@ -0,0 +1,805 @@
+        )
+
+    def call(self, inputs):
+        return keras.ops.matmul(inputs, self.w) + self.b
+
+
+layer = MyCustomLayer(3)
+data = np.random.uniform(size=[3, 3])
+model = keras.models.Sequential([layer])
+model.compile(optimizer="adam", loss="mse")
+model.fit(data, data)
+# Verify that the variables are now being tracked
+for layer in model.layers:
+    print(layer.trainable_variables)
+
+"""
+### None entries are not allowed as part of nested (e.g. list/tuples) tensor arguments in
+Layer.call(), nor as part of call() return values.
+
+The following snippet of code will reproduce the error.
+
+```
+class CustomLayer(keras.layers.Layer):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, inputs):
+      return  inputs["foo"]
+
+layer = CustomLayer()
+inputs = {
+    "foo": keras.Input(shape=(1,), name="foo"),
+    "bar": {
+        "baz": None,
+    },
+}
+layer(inputs)
+```
+"""
+
+"""
+Here is how you will fix it:
+
+Replace `None` with a value or Keras Tensor
+"""
+
+
+class CustomLayer(keras.layers.Layer):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, inputs):
+        output = inputs["foo"]
+        return None
+
+
+layer = CustomLayer()
+inputs = {
+    "foo": keras.Input(shape=(1,), name="foo"),
+    "bar": {
+        "baz": keras.Input(shape=(1,), name="bar"),
+    },
+}
+layer(inputs)
+
+"""
+### Removed features
+
+
+1. Symbolic `add_loss()`: Symbolic `add_loss()` is removed (you can still use
+`add_loss()` inside the `call()` method of a layer/model).
+2. Locally-connected layers: Locally-connected layers are removed due to low usage. To
+use locally-connected layers, copy the layer implementation into your own codebase.
+3. Kernelized layers: Kernelized layers are removed due to low usage. To use kernelized
+layers, copy the layer implementation into your own codebase.
+4. Removed layer attributes: Layer attributes `metrics`, `dynamic` are removed
+5. RNN layer args: The `constants` and `time_major` arguments in RNN layers are removed.
+The `constants` argument was a remnant of Theano and had very low usage. The `time_major`
+argument was also infrequently used.
+6. reset_metrics argument: The reset_metrics argument is removed from `model. *_on_batch`
+methods. This argument had very low usage.
+7. RadialConstraint: The RadialConstraint constraint object is removed. This object had
+very low usage.
+"""
+
+"""
+## Switching Tensorflow code to backend agnostic keras 3.0
+
+Keras 3.0 code with the TensorFlow backend will work with native TensorFlow APIs.
+However, if you want your code to be backend-agnostic, you will need to replace all of
+the `tf.*` API calls with their equivalent Keras APIs.
+
+Follow these instructions to migrate your existing TensorFlow code to Keras 3.0 and run
+it with any backend of your choice :
+
+1. Update imports : replace from tensorflow import keras to import keras
+2. Update code : replace `tf.keras.*` to `keras.*`. 99% of the tf.keras.* API is
```

Comment:
Same here, this won't read well when `tf.keras` is `keras` 3.

Reply:
updated the description